devices/src/virtio/queue.rs

```diff
@@ -322,9 +322,6 @@ impl Queue {
 
     /// A consuming iterator over all available descriptor chain heads offered by the driver.
     pub fn iter<'a, 'b>(&'b mut self, mem: &'a GuestMemory) -> AvailIter<'a, 'b> {
-        if !self.is_valid(mem) {
```

Comment:
I'm not 100% sure about this. Your logic is right, but we need to be very confident there is no way to alter any of these queue params after device activation, otherwise removing this opens up a security hole.

Other reviewers, please think hard on this part.

Reply:
I've reviewed all accesses of fields validated by this function and they are indeed not modified anywhere after `VirtDevice::activate()` :+1: